### ISSUE TITLE
EXTERNAL_PROPERTY adds property multiple times and in multiple places

### ISSUE THREAD

**Issue by thatsnotright** [FasterXML/jackson-databind#222]
I created an example of what I think the bug is here:

https://github.com/thatsnotright/jackson-bug

What I want is a class called Widget that has a WidgetConfig property.  This WidgetConfig is, by default, extensible to any JSON properties which are added.  However there are certain cases where I need a very specific set of properties extracted, but the others I do not care about.

Perhaps I am misunderstanding the use of EXTERNAL_PROPERTY.

Thanks,
Rob

**Comment by cowtowncoder** [comment]
Alas, while the fix would be conceptually simple here:

```
when constructing `TypeSerializer` for `EXTERNAL_PROPERTY`, see if an existing property exists for parent POJO Type serializer; if so, either mark it as no-op, or perhaps even just return `null` for serialization
```

actual implementation requires big changes to internal plumbing; probably mostly in `BeanSerializerFactory`, method `_constructWriter()` and/or `findPropertyTypeSerializer()`.

And that seems like it needs to wait until 2.6 is released.

**Comment by cowtowncoder** [comment]
One additional note: the way `BeanSerializerFactory` work is sub-optimal. Instead of being stateless, passing (too) many arguments around, should really have a stateful builder that collects information.
This is done at earlier point, with `POJOPropertiesCollector`, but factories are leftovers from Jackson 1.0 or so.

**Comment by cowtowncoder** [comment]
Amusing how things work. I was ready to postpone, but then figured out different angle for solving the issue, and that works. So fix is in `master`, to be included in the official 2.6, one of oldest squashed bugs.
